app/code/Vaimo/QtyManager/view/frontend/web/js/qty-manager.js

```diff
@@ -0,0 +1,150 @@
+            this.block = this.element.next('[data-qty-wrapper]');
+            this.block.find('[data-input]').replaceWith(this.element);
+            this.qtyInput = this.block.find('.input-text.qty');
+            this.decreaseBtn = this.block.find('[data-qty=decrease]');
+        },
+
+        /**
+         * @description Method for getting current qty value.
+         * @return {number} - quantity from input
+         */
+        _getQnty: function () {
+            return +this.qtyInput.val();
+        },
+
+        /**
+         * @description Method for setting new qty value.
+         * @param {qty} qty - The new quantity value.
+         */
+        _setQty: function (qty) {
+            this.qtyInput.val(qty);
+            this.triggerChangeInput();
+        },
+
+        /**
+         * @description Method for update qty value and total price.
+         * @param {qty} qty - The new quantity value.
+         */
+        _updateQty: function (qty) {
+            this._setQty(qty);
+            // Apply triggerChangeInput on minicart
+            if (this.qtyInput.closest('#mini-cart').length) {
+                this.triggerChangeInput();
+            }
+        },
+
+        /**
+         * @description Method for decrease and increase events.
+         */
+        _bind: function () {
+            let self = this;
+            this.block.on('click', '[data-qty=decrease]', function (e) {
+                e.preventDefault();
+                self._decreaseQty();
+            });
+            this.block.on('click', '[data-qty=increase]', function (e) {
+                e.preventDefault();
+                self._increaseQty();
+            });
+            this.block.on('focusout', '.input-text.qty', function () {
+                if(!this.closest('#mini-cart')){
+                    self._isValidQtyUpdate();
+                }
+            });
+        },
+
+        /**
+         * @description Method for decrease quantity.
+         */
+        _decreaseQty: function() {
+            let newQty = this._getQnty() - 1;
+            if (newQty <= 1) {
+                newQty = 1;
+                this._disableButton();
+            }
+            this._updateQty(newQty);
+        },
+
+        /**
+         * @description Method for increase quantity.
+         */
+        _increaseQty: function() {
+            let newQty = this._getQnty() + 1;
+            if (newQty > 1) {
+                this._enableButton();
+            }
+            this._updateQty(newQty);
+        },
+
+        /**
+         * @description Method for validate quantity.
+         */
+        _isValidQtyUpdate: function () {
+            let newQty = this._getQnty();
+            if (newQty <= 1) {
+                newQty = 1;
+            }
+            this._updateQty(newQty);
+        },
+
+        /**
+         * @description Method for add disabled attribute for decrease button.
+         */
+        _disableButton: function() {
```

Comment:
Better to name it _disableDecreaseButton, because _disableButton more abstract name.